plugins/woocommerce-admin/client/products/hooks/use-product-auto-save.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { debounce, pick } from 'lodash';
+import { Product } from '@woocommerce/data';
+import { useDebounce } from '@wordpress/compose';
+import { useEffect, useMemo, useState } from '@wordpress/element';
+import { useFormContext } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { useProductHelper } from '../use-product-helper';
+
+export const useProductAutoSave = ( dependencies: ( keyof Product )[] ) => {
+	const [ isAutoSaving, setIsAutoSaving ] = useState( false );
+	const { isValidForm, setValue, values } = useFormContext< Product >();
+	const { createOrUpdateProductWithStatus } = useProductHelper();
+
+	const dependencyValues = dependencies.reduce(
+		( acc: unknown[], key: keyof Product ) => {
+			acc.push( values[ key ] );
+			return acc;
```

Comment:
What does `acc` stand for?

Reply:
`acc` is shorthand for `accumulator`.  This was used because when reducing the variable name is commonly the same as the accumulator name, forcing us to create contrived names to avoid conflicts.

I updated this to `dependenciesAccumulator` in https://github.com/woocommerce/woocommerce/pull/36660/commits/76c47d35fc1504cc84b6da79b1d4c3bb7960cf2f.  Let me know if that sounds more clear to you.